### DISCUSSION TITLE
Ability for authors to turn on/off comments?

### DISCUSSION THREAD

**Discussion by jessleenyc** [forem/forem#15317] [Feature Requests]
### User Story / Description
When a post is controversial or getting heated, authors might want to turn off comments. Some open ended questions:
- When comments are off, do existing comments still show? Is it time to revisit whether or not authors should be able to delete comments from their posts?
- Where we we add that 'switch'?

**Comment by frederikcreemers** [comment]
I know this complicates matters, but what if we had the distinction between "locked" comments and "disabled" comments?

- *Locking* prevents anyone (possibly excluding the author and/or mods) to post new comments
- *Disabling* hides the entire comment section.

This would allow the user to disable comments if they don't want t enter a discussion around an article, and lock it if it gets out of hand, with nothing useful being said.
I think it might also make sense to allow mods of a certain level t lock comments.

**Comment by jwoertink** [comment]
To see other examples, you can look at how [reddit](https://www.reddit.com/r/pics/comments/90xkme/to_combat_a_new_massive_spamming_effort_we_are/) and [github](https://github.com/ariya/phantomjs/issues/15345) both handle locking threads. In these cases, you can still see the comments that have been posted already, but no one is able to add new comments, or replies. My guess is that you wouldn't be able to update these had you made a comment already, but I'm not sure on that..

As for _where_ to put this option, I think right in the post where the published option is.

```
---
title: Some cool title
published: true
allow_comments: false 
description: How to not write bad blog posts
tags: programming, lifestyle
---
```

**Comment by benhalpern** [comment]
Another possible comment state I've thought about: Only followers.

So someone could limit comments to people already following them when they made the post. Just a thought to keep the conversation limited to people already keeping up with them.

We currently have a boolean for whether a post even has comments at all, which is an option on the backend.

Ultimately we want to do everything we can to foster good healthy comments, but as we scale there are a few additional options we could be able to turn to.

**Comment by frederikcreemers** [comment]
Hmm, limiting by followers with a time cutoff seems a bit arbitrary to me.
What about limiting it to the people the author follows (implying they
trust them)?

**Comment by tterb** [comment]
This is just my two cents, but as someone who was drawn to the community based on the constructive and friendly discussion that can be found in the majority of posts, I fear that providing the ability to remove comments entirely may project a misleading impression of the community and hinder the potential for constructive discourse.  
I think that the addition of the "trusted member" system is a good example of how moderation can be handled without overburdening the dev team or a select group of moderators or requiring preemptive restrictions. Though, I think that adding the ability for discussions to become locked if it is deemed as toxic by multiple trusted members or moderators could add additional protection from negative threads proliferating on the platform.

**Comment by thejessleigh** [comment]
I like the idea of being able to turn off new comments on a post I wrote after a given period of time. I'm thinking of things like Sandi Metz's AMA. Right now, she just has [CLOSED] in the title. Being able to disable new comments is great for formats that are intended to be short lived, like AMAs, or news specific discussion items that might lose relevance after a while.

**Comment by stale** [comment]
Thanks for contributing to this issue. As it has been 90 days since the last activity, we are automatically closing the issue in 7 days. This is often because the request was already solved in some way and it just wasn't updated or it's no longer applicable. If this issue still requires attention, please respond with a comment. Happy Coding!

**Comment by frederikcreemers** [comment]
This would still be useful.

**Comment by WesSouza** [comment]
I'd really like to turn comments off on all my posts.

I like to keep the content static, and have conversations on other means such as Twitter.

My ideal situation would be something like:

```yaml
comments_enabled: false
comments_twitter_hashtag: MyPostHashtag
```

Which would then show this below the author information, with a nice design:

```html
<a href="https://twitter.com/hashtag/MyPostHashtag" class="cute-button">
  Join the conversation on Twitter
</a>
```

**Comment by stale** [comment]
Thanks for contributing to this issue. As it has been 90 days since the last activity, we are automatically closing the issue in 7 days. This is often because the request was already solved in some way and it just wasn't updated or it's no longer applicable. If this issue still requires attention, please respond with a comment. Happy Coding!

**Comment by maestromac** [comment]
not stale

**Comment by ryanlanciaux** [comment]
I'm wondering if there's been any more thought on this one. Even when it's not a case of controversial, or heated discussion, it'd be nice to be able to disable certain comments as a post author. I understand that one of the goals for dev.to is foster community and disabling conversation doesn't really move that goal forward. 

That said, it's pretty unfortunate on posts that are announcing a new open source tool or something along those lines, but the comments turn in to ads for everything else that's in a similar space.

**Comment by cherscarlett** [comment]
I would really like to see this implemented.

I'll reiterate what others have stated:

- The author may not want comments from the get-go, and there's plenty of folks who may have great content to contribute that are reluctant to do so because they would be unable to handle any feedback at all. This can be for a variety of reasons that aren't being unwilling to hear ~feedback~ constructive criticism or healthy pushback. 
- The conversation may have gotten too heated, too fast and moderators may need to take the time to clear out destructive comments and resolve issues marked unnecessarily as  "low quality".
- The conversation may have gotten too heated and to continue any discussion at all would serve no constructive end.
- The article may be out of date and commentary may be inappropriate.
- The comments section may be overwhelmed by "ads" on announcements about new OSS tools or similar posts.

**Comment by ellativity** [comment]
[This has been implemented](https://github.com/forem/forem/pull/13808) - @cmgorton what happens to this Discussion now?

**Reply by cmgorton** [reply]
Thanks for letting me know @ellativity.  I will turn it into an issue and we can close it 🥳

**Reply by jeremyf** [reply]
It doesn't look like the PR was merged, but instead closed without merge.  So I believe it's still a "to complete" task
